homeassistant/components/tessie/entity.py

```diff
@@ -43,3 +46,28 @@ def __init__(
     def _value(self) -> Any:
         """Return value from coordinator data."""
         return self.coordinator.data[self.key]
+
+    def get(self, key: str | None = None, default: Any | None = None) -> Any:
+        """Return a specific value from coordinator data."""
+        return self.coordinator.data.get(key or self.key, default)
+
+    async def run(
+        self, func: Callable[..., Awaitable[dict[str, bool]]], **kargs: Any
+    ) -> bool:
+        """Run a tessie_api function and handle exceptions."""
+        try:
+            response = await func(
+                session=self.coordinator.session,
+                vin=self.vin,
+                api_key=self.coordinator.api_key,
+                **kargs,
+            )
+        except ClientResponseError as e:
+            raise HomeAssistantError from e
+        return response.get("result") is True
```

Comment:
It would be better to control the flow with either exceptions or booleans. 

Maybe this should raise if it fails instead of returning True/False ?

Reply:
So I have gone back and looked at the API, and I think this check is redundant. The API either returns 200 with "result": True or it returns a non 200 status code and the library uses raise_for_status(). I will remove this logic.